WebDriverAgentLib/Categories/XCUIElement+FBFind.m

```diff
@@ -27,7 +28,7 @@ @implementation XCUIElement (FBFind)
 + (NSArray<XCUIElement *> *)fb_extractMatchingElementsFromQuery:(XCUIElementQuery *)query shouldReturnAfterFirstMatch:(BOOL)shouldReturnAfterFirstMatch
 {
   if (!shouldReturnAfterFirstMatch) {
-    return query.allElementsBoundByIndex;
+    return query.allElementsBoundByAccessibilityElement;
```

Comment:
Isn't this going to break the backward compatibility?

Reply:
Not sure; but ByIndex is horribly fragile. If your query is "all buttons" for example, each element is then the "#3 button". If the elements on the page change, when the snapshots are refreshed the "#3 button" still resolves successfully and you get a different button for your element id.

Reply:
can you please also add this comment here? Also, there might be other places in the code where the same problem could be fixed then.